### DISCUSSION TITLE
How this project compares to dotnet/wcf?

### DISCUSSION THREAD

**Discussion by sc030268** [CoreWCF/CoreWCF#300] [General]
How can we compare this project with https://github.com/dotnet/wcf?
Which more or less features are there?
Thanks

**Comment by Kralizek** [comment]
The description of dotnet/wcf says
```
This repo contains the client-oriented WCF libraries that enable applications built on .NET Core to communicate with WCF services
```
So it only contains the client side of WCF.

This repo, otoh, contains the server side.

**Reply by sc030268** [reply]
Do you mean that this project completes the other? Or can it completely replace that?
I need to port some project with client/server functionnalities and I don't know which to include.
Please forgive my ignorance, thank you!

**Reply by Kralizek** [reply]
They work together.

The client side (dotnet/wcf) is owned by Microsoft. It is compatible with old .net framework servers.

The server side (corewcf/corewcf) is an open source initiative with some kind of microsoft involvement. It is expected to be compatible with old .net framework and dotnet/wcf clients.

So, to answer your question,
- If you have to migrate clients from .net framework to .net core or .net 5, you can use the bits produced by dotnet/wcf
- if you need to migrate your servers from .net framework to .net core or .net 5, you will use the bits produced by this repo.
